pydantic/validators.py

```diff
@@ -486,6 +486,9 @@ def make_literal_validator(type_: Any) -> Callable[[Any], Any]:
     allowed_choices = {v: v for v in permitted_choices}
 
     def literal_validator(v: Any) -> Any:
+        if not isinstance(v, Hashable):
```

Comment:
Wasn't really sure what you prefer here. The alternative to this guard if, would be catching `TypeError`. Less readable, but would only cause overhead in the error case.

Reply:
Catching `TypeError` as well, would also remove the more or less duplicated core for raising the error. I changed it to catch a `TypeError` now.